diff --git a/drivers/i2c/busses/i2c-designware-master.c b/drivers/i2c/busses/i2c-designware-master.c
index 85dbd0eb5392..26dab90bba3d 100644
--- a/drivers/i2c/busses/i2c-designware-master.c
+++ b/drivers/i2c/busses/i2c-designware-master.c
@@ -1073,6 +1073,15 @@ int i2c_dw_probe_master(struct dw_i2c_dev *dev)
 		dev_err(dev->dev, "failure adding adapter: %d\n", ret);
 	pm_runtime_put_noidle(dev->dev);
 
+	/* recovery the i2c bus */
+	ret = i2c_dw_acquire_lock(dev);
+	if (ret)
+		return ret;
+	dev_err(dev->dev, "recovery the i2c bus\n");
+	i2c_recover_bus(&dev->adapter);
+	i2c_dw_init_master(dev);
+	i2c_dw_release_lock(dev);
+
 	return ret;
 }
 EXPORT_SYMBOL_GPL(i2c_dw_probe_master);
